lib/synapse/haproxy.rb

```diff
@@ -1089,18 +1103,10 @@ def update_backends(watchers)
           end
 
           # actually write the command to the socket
-          begin
-            s = UNIXSocket.new(@opts['socket_file_path'])
-            s.write(command)
-            output = s.read()
-          rescue StandardError => e
-            log.warn "synapse: unknown error writing to socket"
+          output = haproxy_exec(command)
+          unless output == "\n"
+            log.warn "synapse: socket command #{command} failed: #{output}"
```

Comment:
although this is unrelated to the main goal of the PR, i would love to see these log lines follow the `synapse: restart required` format we use elsewhere in this function.

```ruby
log.warn "synapse: restart required because socket command #{command} failed with out #{output}"
```